packages/gcf-utils/src/gcf-utils.ts

```diff
@@ -146,35 +180,97 @@ export class GCFBootstrapper {
     return config as Options;
   }
 
+  parseRequestHeaders(
+    request: express.Request
+  ): {name: string; id: string; signature: string; taskId: string} {
+    const name =
+      request.get('x-github-event') || request.get('X-GitHub-Event') || '';
+    const id =
+      request.get('x-github-delivery') ||
+      request.get('X-GitHub-Delivery') ||
+      '';
+    // TODO: add test to validate this signature is used on the initial
+    // webhook from GitHub:
+    const signature =
+      request.get('x-github-delivery') ||
+      request.get('X-GitHub-Delivery') ||
+      '';
+    const taskId =
+      request.get('X-CloudTasks-TaskName') ||
+      request.get('x-cloudtasks-taskname') ||
+      '';
+    return {name, id, signature, taskId};
+  }
+
+  parseTriggerType(name: string, taskId: string): TriggerType {
+    if (
+      !taskId &&
+      (name === 'schedule.repository' || name === 'pubsub.message')
+    ) {
+      return TriggerType.SCHEDULER;
+    } else if (!taskId && name) {
+      return TriggerType.GITHUB;
+    } else if (name) {
+      return TriggerType.TASK;
+    }
+    return TriggerType.UNKNOWN;
+  }
+
+  buildTriggerInfo(
```

Comment:
This looks like new public interface - can we add jsdocs?

Also, this can probably be static, private, or live outside the class completely.

Reply:
I can move it to be outside the class/static. Also, this is public just for testing purposes. I can make it private and then access it in tests using JavaScript props. Either way, happy to add JSDocs